### ISSUE TITLE
MRPC Score Lower than Expected

### ISSUE THREAD

**Issue by jplehmann** [huggingface/pytorch-pretrained-BERT#346]
I expect to see MRPC scores between 84-88% as advertised.  What I am seeing  with different seeds is 79-84% consistently.  (I thought perhaps the weight initialization was the issue but seems not to be the case #339.)

I am running with the provided command and fp16, using a GCE instance with a Tesla T4.

> time python run_classifier.py   --task_name MRPC   --do_train   --do_eval   --do_lower_case   --data_dir $GLUE_DIR/MRPC/   --bert_model bert-base-uncased   --max_seq_length 128   --train_batch_size 32   --learning_rate 2e-5   --num_train_epochs 3.0   --output_dir /tmp/mrpc_output18/   --seed 18  --fp16

Example outputs:
```
mrpc_output15/eval_results.txt
eval_accuracy = 0.803921568627451
eval_loss = 0.44585930088571474
global_step = 345
loss = 0.42245775305706523

mrpc_output16/eval_results.txt
eval_accuracy = 0.8455882352941176
eval_loss = 0.38226841594658645
global_step = 345
loss = 0.25925399116847825

mrpc_output17/eval_results.txt
eval_accuracy = 0.7916666666666666
eval_loss = 0.4917685123635273
global_step = 345
loss = 0.24811905570652174

mrpc_output3/eval_results.txt
eval_accuracy = 0.8431372549019608
eval_loss = 0.42019053533965467
global_step = 345
loss = 0.2503709876019022

mrpc_output42/eval_results.txt
eval_accuracy = 0.8406862745098039
eval_loss = 0.44909875124108556
global_step = 345
loss = 0.21309310249660326

mrpc_output44/eval_results.txt
eval_accuracy = 0.8406862745098039
eval_loss = 0.45059946084431574
global_step = 345
loss = 0.10150747223068839

mrpc_output45/eval_results.txt
eval_accuracy = 0.8063725490196079
eval_loss = 0.42597512491777834
global_step = 345
loss = 0.29104428498641305

mrpc_output18/eval_results.txt
eval_accuracy = 0.8161764705882353
eval_loss = 0.4096583215629353
global_step = 345
```

Other output:
```
03/05/2019 18:23:31 - INFO - pytorch_pretrained_bert.modeling -   Model config {                                                       
  "attention_probs_dropout_prob": 0.1,                                                                                                  
  "hidden_act": "gelu",                                                                                                                 
  "hidden_dropout_prob": 0.1,                                                                                                          
  "hidden_size": 768,                                                                                                                   
  "initializer_range": 0.02,                                                                                                           
  "intermediate_size": 3072,                                                                                                           
  "max_position_embeddings": 512,                                                                                                       
  "num_attention_heads": 12,                                                                                                            
  "num_hidden_layers": 12,                                                                                                              
  "type_vocab_size": 2,                                                                                                                 
  "vocab_size": 30522                                      
} 
```
Any ideas on why this is the case?  Happy to provide more output.

**Comment by jplehmann** [comment]
Argh, just realized I was on `0.3.0` which is what pip installed due to some dependencies.  Upgrading to `0.6.1` and now I'm getting expected scores:
```
eval_accuracy = 0.8529411764705882
eval_loss = 0.39120761538837473
global_step = 345
loss = 0.17308216924252717

eval_accuracy = 0.8431372549019608
eval_loss = 0.49456917187746835
global_step = 345
loss = 0.12193756103515625

eval_accuracy = 0.875
eval_loss = 0.4023934503396352
global_step = 345
loss = 0.14832657523777174

eval_accuracy = 0.8553921568627451
eval_loss = 0.44353585865567713
global_step = 345
loss = 0.17814078952955162
```
